scripts/run_tests.sh

```diff
@@ -6,10 +6,11 @@ set -e
 (cd /app/relation_engine_server/test/spec_release && \
   tar czvf spec.tar.gz sample_spec_release)
 
-black .
-flake8 --max-complexity 20 /app
-mypy --ignore-missing-imports /app
-bandit -r /app
+# TODO: why reformatting code?
+# black .
```

Comment:
To ensure that the code has a uniform layout. It assumes that whoever is developing runs the tests regularly before committing code.

Reply:
Yes, I understand what it is doing, but I don't understand why one would want this over, for instanced, a developer choice to format-on save, or running black manually.  As a developer I would not expect a test to modify any code.
I noticed this as an issue because the scope is the entire repo, and if one is using a virtual environment it reformats all dependencies as well (and can get tripped up doing so.)

Reply:
I don't think other users have employed a virtual environment, which is probably why it hasn't come up until now. The main advantage of having black and those various code format checks in the tests is that they are inescapable -- the tests will fail if they don't pass. A more proper place for them would be implemented in the dev's IDE, as git hooks, as some sort of auto-committed GHA, etc.

In lieu of that, can you alter the scope of the black command to limit it to py files from the RE repo, instead of every file in the known world?

Reply:
If you add the command line param `--check` to the black command, it will just test the formatting, but not change it. That seems appropriate.

Reply:
Good idea, but Flake8 error messages are much better than black (it isn't a linter.)

As black compatibility seems to a goal of this repo, it would be better to use the flake8-black plugin which will do the linting and if that passes will see if black would trigger a reformat and report that as a familiar flake8 message.

I've added that change, and also the recommended flake8 settings for black compatibilit (https://black.readthedocs.io/en/stable/compatible_configs.html#flake8), which led to fixing a bunch of files which exceed the black default line length of 88. Most of those were comment lines, but a few were code lines.

Reply:
These changes have made the PR rather large and confusing -- can you remove the last two commits from this PR (`git reset --soft HEAD~2` and then force push), and create a new PR with just the formatting stuff and module updates in it? I will merge that one and then you can rebase this PR off that. Thank you!

Reply:
Ok, I'll try to get to that this week.